### PR TITLE
DOCK-2533: fix regex errors in Edit Info dialog

### PR DESCRIPTION
**Pull request by hyunnaye**
**Description**
This PR fixes the regex errors showing on Firefox when opening the "Edit Info" dialog in the My Tools page. 

These errors show up on FireFox is because they use `v` mode - hence the trailing `/v` on the end of the regex patterns in the errors. This `v` flag is stricter than the `u` flag and required me to escape the characters. More information is [here](https://github.com/tc39/proposal-regexp-v-flag?tab=readme-ov-file#how-is-the-v-flag-different-from-the-u-flag)

Before:
image

After:
image

Verifies the regex still works:
image

**Review Instructions**
1. Go to My Tools
2. Select a tool, then go to versions tab
3. For a version, click Actions | Edit
4. Verify no regex errors are in the console.
5. Also edit the version with an invalid path to verify that the regex still works properly

**Issue**
https://ucsc-cgl.atlassian.net/browse/DOCK-2533

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that your code compiles by running `npm run build`
- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] If this is the first time you're submitting a PR or even if you just need a refresher, consider reviewing our [style guide](https://github.com/dockstore/dockstore/wiki/Dockstore-Frontend-Opinionated-Style-Guide#pr-checklist)
- [x] Do not bypass Angular sanitization (bypassSecurityTrustHtml, etc.), or justify why you need to do so
- [x] If displaying markdown, use the `markdown-wrapper` component, which does extra sanitization
- [x] Do not use cookies, although this may change in the future
- [x] Run `npm audit` and ensure you are not introducing new vulnerabilities
- [x] Do due diligence on new 3rd party libraries, checking for CVEs
- [x] Don't allow user-uploaded images to be served from the Dockstore domain
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.
- [x] Check whether this PR disables tests. If it legitimately needs to disable a test, create a new ticket to re-enable it in a specific milestone.

### CODE REVIEW

**Review by coverbeck — APPROVED** [review]
The PR review instructions should also include a step to verify that the regexes are working properly, e.g., if you enter an invalid extension, does an error show up in the UI?

**Review by denis-yuen — APPROVED** [review]

**Comment by svonworl** [comment]
Could you please add a brief summary why the changes were necessary to the PR "Description"?  If you have links to the appropriate docs, that would be great.

**Comment by hyunnaye** [comment]
> Could you please add a brief summary of why the changes were necessary to the PR "Description"? If you have links to the appropriate docs, that would be great.

Added some explanation in the description but please let me know if it is confusing.

**Comment by svonworl** [comment]
> > Could you please add a brief summary of why the changes were necessary to the PR "Description"? If you have links to the appropriate docs, that would be great.
> 
> Added some explanation in the description but please let me know if it is confusing.

That explains it, thanks!

Some of the other regexps (`repoNameWithSlashesImagePath` and `privateAmazonImagePath`) look like they might include slashes in the character classes, do we need to modify them, too?

I wonder if we should add a test that attempts to use all of the regexps in `v` mode, so we can detect problems before we deploy...

**Review by kathy-t — COMMENTED** [review]
Was here, waiting for Steve's feedback to be addressed

File: src/app/shared/validationMessages.model.ts [hunk 1]
```diff
@@ -67,25 +67,25 @@ export const exampleDescriptorPatterns = {
...
+  gitPath: '^([a-zA-Z0-9]+([-_.][a-zA-Z0-9]+)*)\\/([a-zA-Z0-9]+([-_.][a-zA-Z0-9]+)*)$',
+  smkPath: '^\\/([^\\/?:*\\|<>]++\\/)*(Snakefile|[^.\\/?:*\\|<>]++.smk))$',
+  cwlPath: '^\\/([^\\/?:*\\|<>]+\\/)*[^\\/?:*\\|<>]+.(cwl|yaml|yml)',
+  wdlPath: '^\\/([^\\/?:*\\|<>]+\\/)*[^\\/?:*\\|<>]+.wdl$',
+  nflPath: '^\\/([^\\/?:*\\|<>]+\\/)*[^\\/?:*\\|<>]+.(config)',
+  dockerfilePath: '^\\/([^\\/?:*\\|<>]+\\/)*(([a-zA-Z]+[.])?Dockerfile|Dockerfile([.][a-zA-Z]+)?)$',
+  testFilePath: '^\\/([^\\/?:*\\|<>]+\\/)*[^\\/?:*\\|<>]+.(json|yml|yaml)$',
+  imagePath: '^(([a-zA-Z0-9]+([-_.][a-zA-Z0-9]+)*)|_)\\/([a-zA-Z0-9]+([-_.][a-zA-Z0-9]+)*)$',
+  repoNameWithSlashesImagePath: '^(([a-zA-Z0-9]+([-_.][a-zA-Z0-9]+)*)|_)\\/([a-zA-Z0-9]+([-_./][a-zA-Z0-9]+)*)$',
+  privateAmazonImagePath: '^_\\/([a-zA-Z0-9]+([-_./][a-zA-Z0-9]+)*)$', // Has an empty namespace. Allows for slashes in repo name
```
Comment (svonworl) [inline src/app/shared/validationMessages.model.ts:79]: Needs an escape on the slash in the character class (inside the square brackets)?

**Comment by svonworl** [comment]
> Added some tests. Should I add more?

Maybe a test that checks that each value in `validationDescriptorPatterns` is valid in `v` mode?  Something like:
```
    for ([name, pattern] of Object.entries(validationDescriptorPatterns))
        new RegExp(pattern, 'v');
    }
```
(Note that the above might not be the best Typescript, and might be good to tweak so that if a pattern fails, we know which one.)

Curious what others think.

**Review by svonworl — APPROVED** [review]

**Review by kathy-t — APPROVED** [review]

**Pull request merged**